### PR TITLE
Metrics setup

### PR DESCRIPTION
**Pull request by cooldracula**
This PR brings in an initial metrics role, intended to set up the prometheus, grafana, and alertmanager services (like at prometheus.planetary.tools)

To make it work, we needed to generalise how the certbot-cloudflare role sets domains, so it can work for when you have multiple domains on a single host.

There is also a small addition of a flake.nix, for setting up a repo-specific dev environment. It only comes into play when you are using nix + direnv, and has no real impact on the code itself.

### CODE REVIEW

File: roles/nginx/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+- name: Install nginx
+  become: true
+  ansible.builtin.apt:
+    pkg:
+      - nginx
+    state: latest
+
+
+- name: etc/hosts - Prevent nameclashes with hostname
```
Comment (mplorentz) [inline roles/nginx/tasks/main.yml:19]: This looks like it would've been hard to figure out.

File: roles/metrics/templates/nginx/prometheus.conf.tpl
Comment (mplorentz) [inline roles/metrics/templates/nginx/prometheus.conf.tpl:1]: It hurts a little bit to see all this nginx conf repeated with minor changes. But it's probably not worth trying to combine them more, unless there's a really easy way to do it.

File: metrics-example-inventory.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# An example inventory for use with the metrics playbook
+
+all:
+  vars:
+    admin_username: admin
+    admin_password: bcrypthashed_password
```
Comment (mplorentz) [inline metrics-example-inventory.yml:6]: is it time for proper secret management?

File: flake.nix
Comment (mplorentz) [inline flake.nix:1]: I think I understand. It's like a Dockerfile but not.

**Review by mplorentz — APPROVED** [review]
This looks very nice. Feel free to merge when you are ready. I left some minor feedback but it doesn't necessarily mean changes are needed so I'm clicking Approve.

File: roles/metrics/templates/nginx/prometheus.conf.tpl
Comment (cooldracula) [reply roles/metrics/templates/nginx/prometheus.conf.tpl:1]: I did a slight adjustment to the nginx template and reduced it to a single step.  Fairly easy to do, once I got into the swing of the jinja syntax.

**Pull request merged**